fitbenchmarking/fitbenchmarking
# 135 make all plots

#### Description of Work

Fixes #135 

Restructures outputting so that resulting files are created in the following order:
 - All directories
 - All plots
 - All support pages
 - All tables
 - Index page

Adds links from table cells to a connected support page.

Refactors plotting to remove repeated calculations.


#### Testing Instructions

1. Test table looks as it should, links work, and plots are correct on the supporting page

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
